test/integration/targets/pids/tasks/main.yml

```diff
@@ -0,0 +1,52 @@
+# Test code for the pid module
+# Copyright 2018, Saranya sridharan
+
+- name: "Installing PidOf"
+  yum:
+    name: sysvinit-tools
+    state: present
+  when: ansible_distribution == 'CentOS'
+
+- name: "Checking empty result" 
+  pids: 
+    name: "blahblah"
+  register: emptypids
+  
+  
+- name: "Verify the Process IDs (PIDs) returned is empty"
+  assert:
+    that:
+    - emptypids is not changed
+    - emptypids.pids == [] 
+
+  
+  
+- name: Copying the python script
+  copy: 
+    src: sample.py
+    dest: /home
+    mode: 0777
+    
+- name: Executing the python script
+  script: /home/sample.py
```

Comment:
I think you'll need to change two things here:

1. Execute the script using [async](https://docs.ansible.com/ansible/devel/user_guide/playbooks_async.html) so that the next task runs before the script exits.
2. Use [setproctitle](https://pypi.org/project/setproctitle/) to set the process name of your script.

Without the 2nd change you won't be able to distinguish between your script and other python processes, including the ones running ansible-playbook and the module.

Reply:
You'll also need to install `setproctitle` in the test so it's available for use.